app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1681,6 +1683,99 @@ impl GraphEditorModel {
         find_free_place(starting_from, direction, node_areas)
     }
 
+    pub fn new_node_position_after_dropping_edge(
```

Comment:
This name describes a behavior not a semantics. Names should be semantically correct. Why I could not use this function after pressing a `+` button or any other button? The name of a function should not explain when it is used (the user of the function decides when to use it!). Name should semantically desccribe what the function does.

Reply:
> This name describes a behavior not a semantics

This is just not true. Describing behavior is explaining the function's body. So the "behavior" name would be "new_node_position_based_on_mouse_and_position_taken_from_source_of_the_edge"

It also does not necessarily restrict usage to one place: we could check in many places what node position "would" be after dropping edge. 

Additionally, when we decide some day that we want to change the position after dropping edge, this would be the function we'd modify. That's only make me more sure that the name is good: the name describes what the function returns.

> Why I could not use this function after pressing a + button or any other button

Because no other button provide and edge_id.